app/main.py

```diff
@@ -1,2 +1,74 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+
+CAPACITY = 8
+TREASHOLD = 2 / 3
+
+
+@dataclass
+class Pair:
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = CAPACITY) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: list[Pair | None] = [None] * self.capacity
+
+    def obtain_index(self, key: Hashable) -> None:
```

Comment:
The `obtain_index` method should return an `int` instead of `None` since it returns an index value.